pkg/reconciler/managed/reconciler.go

```diff
@@ -677,6 +741,19 @@ func (r *Reconciler) Reconcile(_ context.Context, req reconcile.Request) (reconc
 		managed.SetConditions(xpv1.ReconcileError(errors.Wrap(err, errReconcileConnect)))
 		return reconcile.Result{Requeue: true}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 	}
+	defer func() {
+		if disconnectErr := r.external.Disconnect(ctx); disconnectErr != nil {
+			log.Debug("Cannot disconnect from provider", "error", disconnectErr)
+			record.Event(managed, event.Warning(reasonCannotDisconnect, disconnectErr))
+			disconnectErr = errors.Wrap(disconnectErr, errReconcileDisconnect)
+			managed.SetConditions(xpv1.ReconcileError(disconnectErr))
+			if err != nil {
+				err = errors.Wrap(err, disconnectErr.Error())
+			} else {
+				err = disconnectErr
+			}
+		}
+	}()
```

Comment:
I think this block changed since last time I looked. Is the idea that we're trying to avoid squashing the original error in cases where the deferred function is triggered by a case in which we'd return an error? I think it's probably safe to just log and emit an event for disconnect errors without ever returning them - it's not like requeueing (which happens when we return an error) would help recover from not being able to disconnect.